.circleci/config.yml

```diff
@@ -26,18 +26,83 @@ jobs:
           paths:
             - node_modules
           key: v1-dependencies-{{ checksum "package.json" }}
-        
+
       - run:
           name: unit tests
           command: npm test
-          
+
       - run:
           name: coverage reporting
           command: npm run coverage:submit
 
-      - deploy:
-          name: publish
-          command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - middy/*
+
+  build_next:
+    docker:
+      - image: circleci/node:8.10
+
+    working_directory: ~/middy
+
+    # These are almost the same set of steps as above.
+    # Yaml doesn't allow you to concatenate lists: https://github.com/yaml/yaml/issues/35
+    # Not in these steps:
+    #  - uploading code coverage
+    #  - saving workspace for the deploy job
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "package.json" }}
```

Comment:
What about making a cache key different for different node version? Like `v1-dependencies-node8-{{ checksum "package.json" }}` ?

Reply:
does it matter? I guess I don't know enough about node here - does the set of packages differ between v6 and v8?

Reply:
Set of packages no, but there is `npm install` , and node 6 and 8 have different versions of  `npm`  bundled which has different logic of installing dependencies, so file structure will differ so cache won't be very helpful if dependencies need to be reinstalled anyway

Reply:
Thanks for that info! I'll go ahead and make that change.

Reply:
I've made the change, I wasn't sure what to default to - in this case it'll default to `v1-dependencies-`; but I think that doesn't make sense since if the checksum process fails you'll still end up with a cache key of `v1-dependencies-vX-`. However, looking at the [caching docs on circleci](https://circleci.com/docs/2.0/caching/#full-example-of-saving-and-restoring-cache) the pattern is what I've implemented.. thoughts?

Reply:
Cool! Thanks for taking this into account! Defaulting to `v1-dependencies` as a last resort can remain as is, however, I doubt it will be used more than 1 time at all (prob only the very first run after merging this PR) so LGTM

Reply:
Yeah, actually it helped speed up one of my builds right after asking the question :)